File: Exercism/python/kindergarten-garden/kindergarten_garden.py
class Garden(object):
    """Determines which plants each child in the kindergarten class is
    responsible for."""

    def __init__(self, diagram: str, students: list = None):
        children = ['Alice', 'Bob', 'Charlie', 'David', 'Eve',
                    'Fred', 'Ginny', 'Harriet', 'Ileana', 'Joseph',
                    'Kincaid', 'Larry']
        seeds = {
            'C': 'Clover',
            'G': 'Grass',
            'R': 'Radishes',
            'V': 'Violets'
        }
        [row1, row2] = diagram.splitlines()
        self.garden = {
            child: [seeds[p] for p in plants]
            for child, plants in zip(
                sorted(students or children),
                zip(*[iter(row1)]*2 + [iter(row2)]*2)
            )
        }

    def plants(self, name: str) -> list:
        """Lists all plants for a specific child."""
        return self.garden[name]

File: Exercism/javascript/kindergarten-garden/kindergarten-garden.js
'use strict'

const STUDENTS = ['Alice', 'Bob', 'Charlie', 'David',
                  'Eve', 'Fred', 'Ginny', 'Harriet',
                  'Ileana', 'Joseph', 'Kincaid', 'Larry']

const PLANTS = new Map()
  .set('C', 'clover')
  .set('G', 'grass')
  .set('R', 'radishes')
  .set('V', 'violets')

export class Garden {
  constructor(diagram, students = STUDENTS) {
    students.sort()
    diagram.split('\n')
      .map(row => row.match(/.{2}/g))
      .reduce((row1, row2) => row1
        .map((plants, i) => [...plants, ...row2[i]]
          .map(plant => PLANTS.get(plant))))
      .forEach((garden, i) => this[students[i].toLowerCase()] = garden)
  }
}
